public/apps/configuration/utils/password-strength-bar.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiProgress, EuiText } from '@elastic/eui';
+import React from 'react';
+import zxcvbn from 'zxcvbn';
+
+interface PasswordStrengthBarProps {
+  password: string;
+}
+
+export const PasswordStrengthBar = (props: PasswordStrengthBarProps) => {
+  const { password } = props;
+  const passwordStrength = zxcvbn(password);
+  const strength = passwordStrength.score;
+  let message;
+  switch (strength) {
+    case 0:
+      message = 'Very weak';
```

Comment:
nit: may need to add i18n support for all these msgs. But we can do that as a follow-up.

Reply:
Sure, let me create an issue. I think there is a decent amount of text across the repo that is not translated. However, I do not know how to replicate this/change the OSD to use a different default language, so I may need to look into that.

Reply:
Created issue here: https://github.com/opensearch-project/security-dashboards-plugin/issues/1886